src/GogeDao.sol

```diff
@@ -1,1087 +1,793 @@
-        string description;
-        uint256 startTime;
-        uint256 endTime;
-        uint256 amount;  
-    }
+    /// @notice Utility function so this contract can accept BNB.
+    receive() payable external {}
 
-    struct UpdateMinimumBalanceForDividends {
-        string description;
-        uint256 startTime;
-        uint256 endTime;
-        uint256 amount;  
-    }
+    /// @notice is used to create a new poll.
+    /// @param  _pollType enum type of poll being created.
+    /// @param  _change the matching metadata that will result in the execution of the poll.
+    function createPoll(PollType _pollType, Proposal memory _change) external {        
+        require(createPollEnabled, "GogeDao.sol::createPoll() Ability to create poll is disabled");
+        if (msg.sender != owner) require(getActivePollsFromAuthor(msg.sender) < maxPollsPerAuthor, "GogeDao.sol::createPoll() Exceeds maxPollsPerAuthor");
+        require(block.timestamp < _change.endTime, "GogeDao.sol::createPoll() End time must be later than start time");
+        require(_change.endTime - block.timestamp >= minPeriod, "GogeDao.sol::createPoll() Polling period must be greater than minPeriod");
+        require(_change.endTime - block.timestamp <= maxPeriod, "GogeDao.sol::createPoll() Polling period must be less than maxPeriod");
 
-    struct ModifyBlacklist {
-        string description;
-        uint256 startTime;
-        uint256 endTime;
-        address payable addr;
-        bool blacklisted;
-    }
+        require(IGogeERC20(governanceTokenAddr).balanceOf(msg.sender) >= minAuthorBal, "GogeDao.sol::createPoll() Insufficient balance of tokens");
+        require(IGogeERC20(governanceTokenAddr).transferFrom(msg.sender, address(this), minAuthorBal));
 
-    struct TransferOwnership {
-        string description;
-        uint256 startTime;
-        uint256 endTime;
-        address payable addr;
-    }
+        pollNum += 1;
 
-    struct MigrateTreasury {
-        string description;
-        uint256 startTime;
-        uint256 endTime;
-        address payable addr;
-        address token;
-    }
+        emit ProposalCreated(pollNum, _pollType, _change.endTime);
 
-    struct SetQuorum {
-        string description;
-        uint256 startTime;
-        uint256 endTime;
-        uint256 amount;
-    }
+        _addToVoterLibrary(pollNum, msg.sender);
+        _addToAdvocateFor(pollNum, msg.sender);
 
-    struct SetMinPollPeriod {
-        string description;
-        uint256 startTime;
-        uint256 endTime;
-        uint256 amount;
-    }
+        polls[pollNum][msg.sender] += minAuthorBal;
+        totalVotes[pollNum]        += minAuthorBal;
 
-    struct UpdateGovernanceToken {
-        string description;
-        uint256 startTime;
-        uint256 endTime;
-        address addr;
-    }
+        pollTypes[pollNum]     = _pollType;
+        pollMap[pollNum]       = _change;
+        pollAuthor[pollNum]    = msg.sender;
```

Comment:
Each poll has a proposal (defined by data fields Proposal struct). The proposal is contained within the poll, but is not the poll. The poll also contains things like an author which are not directly part of the proposal, but part of the poll. Hope that makes sense

Reply:
I agree that a proposal is contained within a poll but an author creates a proposal, not a poll. A poll is created indirectly as a result of someone creating a proposal, sort of like a side effect lol.

Reply:
Yes that is correct. Do you feel like we should change it to createProposal() instead? Tbh nobody will even see this contract directly since they'll be using a UI to interact with it, but semantically it could make a bit more sense